server/cluster/cluster.go

```diff
@@ -606,6 +607,33 @@ func (c *RaftCluster) HandleStoreHeartbeat(stats *pdpb.StoreStats) error {
 	return nil
 }
 
+// processBucketHeartbeat update the bucket information.
+func (c *RaftCluster) processBucketHeartbeat(buckets *metapb.Buckets) error {
+	c.RLock()
+	region := c.core.GetRegion(buckets.GetRegionId())
+	c.RUnlock()
+	if region == nil {
+		bucketEventCounter.WithLabelValues("region_cache_miss").Inc()
+		return errors.Errorf("region %v not found", buckets.GetRegionId())
+	}
+
+	for i := 0; i < retry; i++ {
```

Comment:
It seems the only thing we want to retry is `UpdateBuckets`? So could we try it directly, the L621-L626 seems shouldn't be included in the retry logic.

Reply:
fix